### ISSUE TITLE
Late state update on unmounted component

### ISSUE THREAD

**Issue by SebBerlioux** [hcikit/hcikit#4]
I receive this warning on Safari when using the createUpload component.

> Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in the componentWillUnmount method.
>     in Upload (created by ConnectFunction)
>     in ConnectFunction
>     in Unknown (created by App)
>     in div (created by DefaultGridLayout)
>     in DefaultGridLayout (created by App)
>     in App (created by ConnectFunction)

**Comment by blainelewis1** [comment]
I kinda know about this one, are you using `fireAndForget` mode? The upload should still be successful so I don't really plan to solve this yet. Thanks for making an issue though it will make it much easier to fix in the future.

**Comment by SebBerlioux** [comment]
yes I all using the `fireAndForget` parameter. It still uploads the logs on every call though.

**Comment by blainelewis1** [comment]
Do you know if this happens every time you use the upload component or only when it fails?

**Comment by SebBerlioux** [comment]
I don't see this warning anymore somehow so I'm not sure
